File: digitalmaktabapi/Models/Enrollment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace digitalmaktabapi.Models
{
    [Table("Enrollment")]
    public class Enrollment : Base
    {
        public required Guid StudentId { get; set; }
        public required Student Student { get; set; }
        public required Guid ClassId { get; set; }
        public required Class Class { get; set; }
        public required Guid CalendarYearId { get; set; }
        public required CalendarYear CalendarYear { get; set; }
    }
}

File: digitalmaktabspa/src/models/Enrollment.ts
import { Base } from "./Base";
import { CalendarYear } from "./CalendarYear";
import { Class } from "./Class";
import { Student } from "./Student";

export interface Enrollment extends Base {
  studentId: string;
  student: Student;
  classId: string;
  class: Class;
  calendarYearId: string;
  calendarYear: CalendarYear;
}
